### PR TITLE
Add new version k8s in CI

### PR DESCRIPTION
**Pull request by kota2and3kan**
## Description

This PR updates the Kubernetes version in the CI.

Also, updated the `Supported Kubernetes Version` section in the `README`. Now, we have an official document that describes `Supported Kubernetes Versions`. So, I updated the `README` to refer to the official document.

In addition, I will update the supported Kubernetes version from `1.26 - 1.30` to `1.27 - 1.31` on the official document side in another PR.

Please take a look!

## Related issues and/or PRs

- https://github.com/scalar-labs/docs-internal-scalardb/pull/421
  - I updated the official document in this PR. 

## Changes made

- Update Kubernetes versions in `.github/workflows/helm_charts_scalar.yml`.
- Update `README.md` to refer to the [official document](https://scalardb.scalar-labs.com/docs/latest/requirements/#kubernetes).

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

Basically, we decide on which Kubernetes versions are supported based on the managed Kubernetes services.

- https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html
- https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli
- https://cloud.google.com/kubernetes-engine/docs/release-notes-regular
- https://docs.oracle.com/en-us/iaas/Content/ContEng/Concepts/contengaboutk8sversions.htm

## Release notes

N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,11 +13,11 @@ This directory contains the following helm charts.
 
 ## Prerequisites
 
-* Helm 3.5+
+You need the [`helm`](https://helm.sh/docs/intro/install/) command. See [Package manager section in Requirements document](https://scalardb.scalar-labs.com/docs/latest/requirements/#package-manager) for details on supported Helm versions.
```
Comment (josh-wong) [inline README.md:16]:
```suggestion
You need the [`helm`](https://helm.sh/docs/intro/install/) command. For details on supported Helm versions, see the [Package manager section in the Requirements document](https://scalardb.scalar-labs.com/docs/latest/requirements/#package-manager).
```

File: README.md [hunk 2]
```diff
@@ -13,11 +13,11 @@ This directory contains the following helm charts.
...
 
 ## Supported Kubernetes versions
 
-* 1.30.x, 1.29.x, 1.28.x, 1.27.x, 1.26.x
+See [Kubernetes section in Requirements document](https://scalardb.scalar-labs.com/docs/latest/requirements/#platform) for details on supported Kubernetes versions.
```
Comment (josh-wong) [inline README.md:20]:
```suggestion
For details on supported Kubernetes versions, see the [Kubernetes section in the Requirements document](https://scalardb.scalar-labs.com/docs/latest/requirements/#platform).
```

File: README.md (hunk 1 above)
Comment (josh-wong) [inline README.md:16]: I think we'll be adding a Requirements doc to the ScalarDL docs. Should we include a "Package manager" section in that doc so that we can link to it here?

File: README.md (hunk 2 above)
Comment (josh-wong) [inline README.md:20]: Similar to my previous comment, I think we'll be adding a Requirements doc to the ScalarDL docs. Should we include a "Platform" section in that doc so that we can link to it here?

**Review by josh-wong — CHANGES_REQUESTED** [review]
I've added some comments and suggestions. PTAL!

File: README.md (hunk 1 above)
Comment (kota2and3kan) [reply README.md]: Ah, yes. I think we should create a `Requirements` document of ScalarDL that includes a `Package manager` section. And, we should link to the document from this README.

File: README.md (hunk 2 above)
Comment (kota2and3kan) [reply README.md]: Yes. I think we should add a `Platform` section in ScalarDL's requirements document and link to it from this README.

**Comment by kota2and3kan** [comment]
@josh-wong 
Thank you for your review!
I applied your suggestions and left several comments.
Please take a look when you have time! 🙇‍♂️

**Review by josh-wong — APPROVED** [review]
I've left one minor suggestion. Other than that, LGTM. Thank you!🙇🏻‍♂️

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
